nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -491,6 +519,15 @@ private boolean isSnapshotOperation(TableMetadata base, TableMetadata metadata)
             || snapshot.snapshotId() != base.currentSnapshot().snapshotId());
   }
 
+  private void addReference(OnReferenceBuilder<?> builder) {
```

Comment:
Call sites would be be simpler, if this would return the properly typed builder back.